runtime/commonMain/src/kotlinx/serialization/SealedSerializer.kt

```diff
@@ -93,11 +93,25 @@ internal class SealedClassSerializer<T : Any>(
     private val serialName2Serializer: Map<String, KSerializer<out T>>
 
     init {
-        require(subclasses.size == subclassSerializers.size) {
-            "Arrays of classes and serializers must have the same length," +
-                    " got arrays: ${subclasses.contentToString()}, ${subclassSerializers.contentToString()}\n" +
-                    "Please ensure that @Serializable annotation is present on each sealed subclass"
+
+        if (subclasses.size != subclassSerializers.size) {
+            // Remove prefix is required for JVM to trim "class " from KClass.toString
+            val allClasses = subclasses.map { it.toString().substringAfter("class ") }
```

Comment:
Did you know that when kotlin-reflect is not in a classpath, `KClass.toString()` looks like this: 
```
class java.lang.String (Kotlin reflection is not available)
```
So you may want to cut out 'reflection is not available' too.
But I think it's better to use `simpleName()`. 

Anyway, what's wrong with original message? I think most of the users can see what classes are missing `@Serializable` annotation because they're all in the same file, so  `Please ensure that @Serializable annotation is present on each sealed subclass / All subclasses of sealed class $bc should be marked @Serializable` is more than enough. Such guessing logic is vague and may give strange results in many cases (e.g. custom serializer that does not contain $$ in classname).

Also, please use `bc.simpleName()` (due to mentioned kotlin-reflect note)

Reply:
Agree, it should be enough. 
Was thinking about providing best-effort guess, but it really seems to be too much